### DISCUSSION TITLE
Mismatch between `definePageMeta` and `extendPages` ?

### DISCUSSION THREAD

**Discussion by qnp** [nuxt/nuxt#16016] [Questions]
Hello,

I want to define page meta in each page using `definePageMeta`, in order to provide meta info as for sitemap generation. I then want to define a Nuxt module that extend pages meta using `extendPages`. I had the following observations:
1. Page meta defined in pages Vue files using `definePageMeta` are **not available** in `pages` handled by module’s `extendPages` hook.
2. Extending `pages` meta in module’s `extendPages` hook will result in overriding page meta defined page-wise using `definePageMeta`.

See this minimal reproduction example (uncomment line 19 in modules/extend-pages/index.ts to see the second observation).
[define-page-meta.zip](https://github.com/nuxt/framework/files/8224464/define-page-meta.zip)

I would have expected:
1. To have `meta` coming from `definePageMeta` populated already in `extendPages` hook.
2. To allow proper extension of pages meta in `extendPages` hook.

Do I miss something, or is it not intended to work like this ?

**Comment by danielroe** [comment]
`definePageMeta` is a compile-time macro, so the page meta isn't available when `extendPages` runs.

**Reply by qnp** [reply]
Ok that’s clear. Thus:
1. Which mechanics do you recommend to access compile-time `pages` meta in a Nuxt module ?
2. Even though, I coined a bug: if we set `meta` for pages in `extendPages` hook, then the meta defined in `definePageMeta` are **not** added to the page. I'll make an issue for that.

**Reply by qnp** [reply]
#3602

**Reply by qnp** [reply]
Fixed by https://github.com/nuxt/framework/pull/3603

**Comment by silverbackdan** [comment]
Hi @qnp and @danielroe . I have a little issue with how we must define the page meta when adding layouts and wondered if you had any ideas?

The layout for a page I am loading is defined by data in a response from an API. This data is stored in a Pinia store. Because I am adding the module first, and then the plugin after, which will add the stores etc. I'm wondering how I could achieve a dynamic value for a page that is being added with `extendPages`. Any thoughts?
